cpp/src/arrow/dataset/file_ipc.cc

```diff
@@ -185,7 +185,13 @@ Result<std::shared_ptr<FileWriter>> IpcFileFormat::MakeWriter(
 
   auto ipc_options = checked_pointer_cast<IpcFileWriteOptions>(options);
 
-  ARROW_ASSIGN_OR_RAISE(auto writer, ipc::MakeFileWriter(destination, schema));
+  // override use_threads to avoid nested parallelism
+  ipc_options->options->use_threads = false;
```

Comment:
Should this in theory only be done if we have multiple files / fragments? (like the issue we had for parquet)

Reply:
That sounds acceptable but I'd like to defer it to a follow up